packages/hardhat/contracts/periphery/lib/ControllerHelperUtil.sol

```diff
@@ -0,0 +1,264 @@
+            _mintAndLpParams.vaultId,
+            _mintAndLpParams.wPowerPerpAmount,
+            0
+        );
+
+        // LP _mintAndLpParams._wPowerPerpAmount & _mintAndLpParams.collateralToLp in Uni v3
+        uint256 uniTokenId = lpWPowerPerpPool(
+            _controller,
+            _nonfungiblePositionManager,
+            _wPowerPerpPool,
+            _wPowerPerp,
+            _vaultId,
+            ControllerHelperDataType.LpWPowerPerpPool({
+                recipient: _mintAndLpParams.recipient,
+                ethAmount: _mintAndLpParams.collateralToLp,
+                amount0Desired: _isWethToken0 ? _mintAndLpParams.collateralToLp : _mintAndLpParams.wPowerPerpAmount,
+                amount1Desired: _isWethToken0 ? _mintAndLpParams.wPowerPerpAmount : _mintAndLpParams.collateralToLp,
+                amount0Min: _mintAndLpParams.amount0Min,
+                amount1Min: _mintAndLpParams.amount1Min,
+                lowerTick: _mintAndLpParams.lowerTick,
+                upperTick: _mintAndLpParams.upperTick
+            })
+        );
+
+        return (_vaultId, uniTokenId);
+    }
+
+    /**
+     * @notice increase liquidityin Uni v3 position
+     * @param _controller controller address
+     * @param _nonfungiblePositionManager Uni NonFungiblePositionManager address
+     * @param _vaultId vault Id
+     * @param _increaseLiquidityParam ControllerHelperDataType.IncreaseLpLiquidityParam struct
+     * @param _isWethToken0 bool variable indicate if Weth token is token0 in Uniswap v3 weth/wPowerPerp pool
+     */
+    function increaseLpLiquidity(address _controller, address _nonfungiblePositionManager, address _wPowerPerp, uint256 _vaultId, ControllerHelperDataType.IncreaseLpLiquidityParam memory _increaseLiquidityParam, bool _isWethToken0) public {
+        if (_increaseLiquidityParam.wPowerPerpAmountToMint > 0) {
+            IController(_controller).mintWPowerPerpAmount{value: _increaseLiquidityParam.collateralToDeposit}(
+                _vaultId,
+                _increaseLiquidityParam.wPowerPerpAmountToMint,
+                0
+            );
+        }
+
+        INonfungiblePositionManager.IncreaseLiquidityParams memory uniIncreaseParams = INonfungiblePositionManager.IncreaseLiquidityParams({
+            tokenId: _increaseLiquidityParam.tokenId,
+            amount0Desired: (_isWethToken0) ? _increaseLiquidityParam.wethAmountToLp : _increaseLiquidityParam.wPowerPerpAmountToMint,
+            amount1Desired: (_isWethToken0) ? _increaseLiquidityParam.wPowerPerpAmountToMint : _increaseLiquidityParam.wethAmountToLp,
+            amount0Min: _increaseLiquidityParam.amount0Min,
+            amount1Min: _increaseLiquidityParam.amount1Min,
+            deadline: block.timestamp
+        });
+
+        INonfungiblePositionManager(_nonfungiblePositionManager).increaseLiquidity(uniIncreaseParams);
+
+        checkExcess(_controller, _nonfungiblePositionManager, _wPowerPerp, _vaultId);
```

Comment:
[#300](https://github.com/opynfinance/squeeth-monorepo/issues/300), can we eliminate this?

Reply:
[#303](https://github.com/opynfinance/squeeth-monorepo/issues/303)can we have ownership check be consistent